components/notifications-service/integration_test/notification_test.go

```diff
@@ -0,0 +1,1207 @@
+			Short: false,
+		}
+		assert.Contains(t, attach.Fields, expectedField1)
+		expectedField2 := &SlackAttachmentField{
+			Value: "Multiple",
+			Title: "Profile",
+			Short: false,
+		}
+		assert.Contains(t, attach.Fields, expectedField2)
+		expectedField3 := &SlackAttachmentField{
+			Value: "chefnode.example",
+			Title: "Node",
+			Short: false,
+		}
+		assert.Contains(t, attach.Fields, expectedField3)
+	})
+}
+
+func TestNotificationDispatchWithOneServiceNowAlertRuleForAllEvents(t *testing.T) {
+	err := suite.DeleteEverything()
+	require.NoError(t, err)
+
+	ts := newTestServer()
+	defer ts.Close()
+
+	// NOTE: the elixir notifications service caches the rules and refreshes them
+	// every 5s, which means we have to wait that long for the rules to take
+	// effect before we test their behavior. To speed things up, we group tests
+	// that can share the same alert rules state.
+	addRules(t,
+		serviceNowAlertRuleFor(api.Rule_CCRFailure, ts.ServiceNowURL(), ""),
+		serviceNowAlertRuleFor(api.Rule_ComplianceFailure, ts.ServiceNowURL(), ""),
+		// Success events are unsupported, which we test for below
+		serviceNowAlertRuleFor(api.Rule_ComplianceSuccess, ts.ServiceNowURL(), ""),
+		serviceNowAlertRuleFor(api.Rule_CCRSuccess, ts.ServiceNowURL(), ""),
+	)
+
+	t.Run("CCRFailure alerts are sent to the ServiceNow URL", func(t *testing.T) {
+		_, err := suite.Client.Notify(ctx, ccrFailureEvent())
+		require.NoError(t, err)
+		postData := ts.GetLastPost()
+		assert.NotNil(t, postData)
+	})
+
+	t.Run("CCRFailure alerts are formatted properly", func(t *testing.T) {
+		_, err := suite.Client.Notify(ctx, ccrFailureEvent())
+		require.NoError(t, err)
+		postData := ts.GetLastPost()
+		assert.NotNil(t, postData)
+
+		// uncomment to see the message the service sent
+		//fmt.Println(string(postData.BodyBytes))
+
+		var actual ServiceNowMessage
+		err = json.Unmarshal(postData.BodyBytes, &actual)
+		require.NoError(t, err)
+
+		assert.Equal(t, "converge_failure", actual.Type)
+		assert.Equal(t, "chefnode.example", actual.NodeName)
+		// AutomateFQDN is populated from the A2 config, so it could be anything.
+		// In the hab studio, it should be a2-dev.test
+		assert.NotEmpty(t, actual.AutomateFQDN)
+		assert.Equal(t, "https://automate.example/nodes/0/runs/1", actual.AutomateFailureURL)
+		assert.Equal(t, "cookbook_example", actual.Cookbook)
+		expectedFailureSnippet :=
+			`Chef client run failure on [https://automate.example/nodes/0] chefnode.example : https://automate.example/nodes/0/runs/1
+Example Exception Occurs
+EXAMPLE EXCEPTION MESSAGE
+(cookbook_example::recipe_example line 13) 
+`
+		assert.Equal(t, expectedFailureSnippet, actual.FailureSnippet)
+		assert.Equal(t, "Example Exception Occurs", actual.ExceptionTitle)
+		assert.Equal(t, "EXAMPLE EXCEPTION MESSAGE\n(cookbook_example::recipe_example line 13)", actual.ExceptionMessage)
+		expectedBT := []string{"/path/to/recipe_example.rb: 42", "/path/to/recipe_example.rb: 23"}
```

Comment:
What is a "BT" ?

Reply:
"backtrace." Seemed fair to abbreviate heavily since we use it just once on the next line